crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -348,52 +348,34 @@ impl<T: TransactionOrdering> TxPool<T> {
-        // tracked
-        match best_transactions_attributes.basefee.cmp(&self.all_transactions.pending_fees.base_fee)
-        {
-            Ordering::Equal => {
-                // for EIP-4844 transactions we also need to check if the blob fee is now lower than
-                // what's currently being tracked, if so we need to include transactions from the
-                // blob pool that are valid with the lower blob fee
-                if best_transactions_attributes
-                    .blob_fee
-                    .is_some_and(|fee| fee < self.all_transactions.pending_fees.blob_fee as u64)
-                {
-                    let unlocked_by_blob_fee =
-                        self.blob_pool.satisfy_attributes(best_transactions_attributes);
-
-                    Box::new(self.pending_pool.best_with_unlocked(
-                        unlocked_by_blob_fee,
-                        self.all_transactions.pending_fees.base_fee,
-                    ))
-                } else {
-                    Box::new(self.pending_pool.best())
-                }
-            }
-            Ordering::Greater => {
-                // base fee increased, we only need to enforce this on the pending pool
-                Box::new(self.pending_pool.best_with_basefee_and_blobfee(
-                    best_transactions_attributes.basefee,
-                    best_transactions_attributes.blob_fee.unwrap_or_default(),
-                ))
-            }
-            Ordering::Less => {
-                // base fee decreased, we need to move transactions from the basefee + blob pool to
-                // the pending pool that might be unlocked by the lower base fee
-                let mut unlocked = self
-                    .basefee_pool
-                    .satisfy_base_fee_transactions(best_transactions_attributes.basefee);
-
-                // also include blob pool transactions that are now unlocked
-                unlocked.extend(self.blob_pool.satisfy_attributes(best_transactions_attributes));
-
-                Box::new(
-                    self.pending_pool
-                        .best_with_unlocked(unlocked, self.all_transactions.pending_fees.base_fee),
-                )
-            }
+        let basefee_decrease =
+            best_transactions_attributes.basefee < self.all_transactions.pending_fees.base_fee;
+        let blobfee_decrease = best_transactions_attributes
+            .blob_fee
+            .is_some_and(|fee| fee < self.all_transactions.pending_fees.blob_fee as u64);
+
+        let mut unlocked = vec![];
+
+        if basefee_decrease {
+            // base fee decreased, we need to move transactions from the basefee to
+            // the pending pool that might be unlocked by the lower base fee
+            unlocked.extend(
+                self.basefee_pool
+                    .satisfy_base_fee_transactions(best_transactions_attributes.basefee),
```

Comment:
not fully following why we need to change it like this,

can we not simply add the missing conditions into the previous match arms?

Reply:
we already have all the functionality that we need, so we should just complete the missing branches with the functions we have for this instead of rewriting it.